src/integration/python/knowrob_module.cpp

```diff
@@ -76,6 +78,33 @@ static inline void register_triple_types() {
 	boost::python::class_<TripleList>("TripleList").def(boost::python::vector_indexing_suite<TripleList, true>());
 }
 
+static void InitKnowledgeBaseWrapper(boost::python::list py_argv) {
+	static int argc = boost::python::len(py_argv);
+	static std::vector<char *> argv;
+
+	for (int i = 0; i < argc; ++i) {
+		std::string arg = boost::python::extract<std::string>(py_argv[i]);
+		argv.push_back(arg.data());
```

Comment:
here you take the pointer to the local string data. so after leaving the for loop the data in the array will eventually be invalid